spark_pipeline_framework/pipelines/framework_file_pipeline.py

```diff
@@ -0,0 +1,133 @@
+        )
+        for transformer in self.transformers:
+            assert isinstance(transformer, FrameworkFileTransformer), type(transformer)
+            stage_name: Optional[str] = None
+            try:
+                i += 1
+                if hasattr(transformer, "getName"):
+                    # noinspection Mypy
+                    stage_name = transformer.getName()  # type: ignore
+                    logger.info(
+                        f"---- Running pipeline [{pipeline_name}] transformer [{stage_name}]  "
+                        f"({i} of {count_of_transformers}) ----"
+                    )
+                else:
+                    stage_name = transformer.__class__.__name__
+                with ProgressLogMetric(
+                    progress_logger=self.progress_logger, name=stage_name or "unknown"
+                ):
+                    self.progress_logger.log_event(
+                        pipeline_name, event_text=f"Running pipeline step {stage_name}"
+                    )
+                    response = transformer.transform(dataset=df, response=response)
+            except Exception as e:
+                if hasattr(transformer, "getName"):
+                    # noinspection Mypy
+                    stage_name = transformer.getName()  # type: ignore
+                else:
+                    stage_name = transformer.__class__.__name__
+                logger.error(
+                    f"!!!!!!!!!!!!! pipeline [{pipeline_name}] transformer [{stage_name}] threw exception !!!!!!!!!!!!!"
+                )
+                # use exception chaining to add stage name but keep original exception
+                friendly_spark_exception: FriendlySparkException = (
+                    FriendlySparkException(exception=e, stage_name=stage_name)
+                )
+                error_messages: List[str] = (
+                    friendly_spark_exception.message.split("\n")
+                    if friendly_spark_exception.message
+                    else []
+                )
+                for error_message in error_messages:
+                    logger.error(msg=error_message)
+
+                if hasattr(transformer, "getSql"):
+                    # noinspection Mypy
+                    logger.error(transformer.getSql())  # type: ignore
+                logger.error(
+                    "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+                )
+                self.progress_logger.log_exception(
+                    event_name=pipeline_name,
+                    event_text=f"Exception in Stage={stage_name}",
+                    ex=e,
+                )
+                raise friendly_spark_exception from e
+        self.progress_logger.log_event(
+            event_name=pipeline_name, event_text=f"Finished Pipeline {pipeline_name}"
+        )
+        return response
+
+    def transform(self, dataset, response, params=None):
```

Comment:
I believe this method, `transform` in this file is redundant, as the transform method call in line 75 is actually defined in line 25-26 in the `FrameworkFileTransformer`.

Reply:
`transform` internally calls the `_transform` function and is not redundant.